### DISCUSSION TITLE
Public key invalid instead of correctly entering

### DISCUSSION THREAD

**Discussion by raheelhp** [emailjs-com/help#109] [Q&A] [answered]
I am entering the correct public key in my Contact.jsx form react but it still shows invalid every time I enter. I have checked my service_id, template_id. everything is correct but it still shows an invalid public key. Please help me out .

**Comment by xr0master** [comment]
If you do everything correctly, it will work.

1. Use the latest version of the SDK.
2. Use the public key directly since many devs do not know how to work with environment variables (aka dotenv)

**Reply by raheelhp** [reply]
the public key on my email-js i am directly using. 
i have imported import emailjs from "emailjs-com";  
emailjs
      .sendForm('service_id', 'template_id', formRef.current, {
        publicKey: 'gNFQa4SZihiiwRucp',
      })
      .then(
        () => {
          console.log('SUCCESS!');
        },
        (error) => {
          console.log('FAILED...', error.text);
        },
      );
  }; see I am directly putting public key here. but when I click on my submit button in the console it shows invalid public key instead it is being the same on email-js dashboard.

**Reply by xr0master** [reply, marked as answer]
The emailjs-com package was deprecated many years ago. Why don't you use the official documentation?

**Reply by raheelhp** [reply]
oh thanks sir done

**Comment by MQasimNazir** [comment]
im entering public key , service id , template id correctly and I've checked many times but it shows the error "400 bad request  , FAILED ... the public key is invalid "     here is my code

 emailjs.sendForm('service_n7lw4xe', 'template_lxq5czq', form.current,
       { publicKey: 'Q05xXYBKniARq09ei',
       }).then(() => {
      console.log('SUCCESS!');
    },
      (error) => {
        console.log('FAILED...', error.text);
      },
    );
    
    
   @xr0master or  anyone else please can guide me what is the problem I'm stucked in this since last  two days ..

**Reply by xr0master** [reply]
Ensure you are using the latest SDK version.

**Reply by Zmirgathon** [reply]
It says the same for mine:
<script type="text/javascript"
        src="https://cdn.jsdelivr.net/npm/@emailjs/browser@4/dist/email.min.js">
  emailjs.init('rXOFQ7UclTJ6rk9gv');
    </script>
    <script>

Even though this is the latest version (or the latest one that they've updated on their site)
